### PR TITLE
feat: strip markdown links in search

### PR DESCRIPTION
**Pull request by Hacksore**
FWIW I did look into alternatives, such as using remark, but since this is a webworker there is no access to the document 😢.

Best I can do is this regex appraoch. 

before:
image

after:
image

### CODE REVIEW

**Comment by w3cj** [comment]
Thanks for the contribution! I think one issue might be:
* If a search term matched a URL, the URL will not be highlighted in the results

Can you check if this is the case?

Maybe instead of stripping the URL entirely, we can reformat a markdown URL to remove the brackets but still include the url.

Like from this:

```
[Hacksore on Twitter](https://twitter.com/Hacksore)
```

To this:


```
Hacksore on Twitter - https://twitter.com/Hacksore
```

File: src/lib/search/search.ts [hunk 1]
```diff
@@ -63,7 +63,14 @@ export function search(query: string) {
 				(a?.block?.breadcrumbs.length || 0) - (b?.block?.breadcrumbs.length || 0) || a.rank - b.rank
 			);
 		})
-		.map(({ block }) => block) as (Block & Show)[];
+		.map(({ block }) => {
+			return {
+				...block,
+				content: block?.content
+					// strip markdown braces but preserve the link text as well as the link, delimited by a hyphen
+					.replace(/\[([^\]]+)\]\(([^)]+)\)/g, '$1 - $2')
```
Comment (Hacksore) [inline src/lib/search/search.ts:71]: not gonna lie copilot cooked this regex so would be ideal to have someone double check it.

Comment (wesbos) [reply src/lib/search/search.ts:71]: prolly fine right

**Comment by wesbos** [comment]
lets get that money! 

I think once the embeddings stuff is up, we can swap the search for that, but this is great for now!

**Pull request merged**